### PR TITLE
LPS-186528 Migrate select fragment collection

### PR DESCRIPTION
**Pull request by BarbaraCabrera**
[Link to ticket](https://liferay.atlassian.net/browse/LPS-186528)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by BarbaraCabrera** [comment]
ci:test:sf

File: modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorViewDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+		Group scopeGroup = themeDisplay.getScopeGroup();
+
+		if ((themeDisplay.getCompanyId() ==
+				PortalInstances.getDefaultCompanyId()) &&
+			scopeGroup.isCompany()) {
+
+			groupIds = ArrayUtil.append(groupIds, CompanyConstants.SYSTEM);
+		}
+
+		long[] allGroupIds = groupIds;
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorViewDescriptor.java:147]: @BarbaraCabrera it is not needed

File: modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorViewDescriptor.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+		_orderByCol = SearchOrderByUtil.getOrderByCol(
+			_httpServletRequest, FragmentPortletKeys.FRAGMENT,
+			"fragment-collections-order-by-col", "create-date");
+
+		return _orderByCol;
+	}
+
+	private PortletURL _getPortletURL() {
+		return PortletURLBuilder.createRenderURL(
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorViewDescriptor.java:209]: @BarbaraCabrera We already have the portletURL reuse it

File: modules/apps/fragment/fragment-web/src/main/resources/META-INF/resources/init.jsp [hunk 3]
```diff
@@ -60,11 +60,8 @@ page import="com.liferay.fragment.web.internal.display.context.GroupFragmentEntr
 page import="com.liferay.fragment.web.internal.display.context.GroupFragmentEntryUsageManagementToolbarDisplayContext" %><%@
 page import="com.liferay.fragment.web.internal.display.context.ImportDisplayContext" %><%@
 page import="com.liferay.fragment.web.internal.display.context.RenderFragmentEntryDisplayContext" %><%@
-page import="com.liferay.fragment.web.internal.display.context.SelectFragmentCollectionDisplayContext" %><%@
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/resources/META-INF/resources/init.jsp:63]: @BarbaraCabrera you should remove also SelectFragmentCollectionDisplayContext.java and SelectFragmentCollectionManagementToolbarDisplayContext.java

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:sf

File: modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/servlet/taglib/util/BasicFragmentEntryActionDropdownItemsProvider.java [hunk 4]
```diff
@@ -390,6 +392,7 @@ private String _getItemSelectorURL() {
 			dropdownItem.putData(
 				"fragmentEntryId",
 				String.valueOf(_fragmentEntry.getFragmentEntryId()));
+
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/servlet/taglib/util/BasicFragmentEntryActionDropdownItemsProvider.java:395]: I think we should remove this empty line

File: modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorViewDescriptor.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+		_orderByCol = SearchOrderByUtil.getOrderByCol(
+			_httpServletRequest, FragmentPortletKeys.FRAGMENT,
+			"fragment-collections-order-by-col", "create-date");
+
+		return _orderByCol;
+	}
+
+	private boolean _isIncludeGlobalFragmentCollections() {
+		if (_includeGlobalFragmentCollections != null) {
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorViewDescriptor.java:197]: You should add it to the criterion, if not when you paginate or click on some parts of the page, this parameter is going to be lost

File: modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorView.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	@Override
+	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
+		return _supportedItemSelectorReturnTypes;
+	}
+
+	@Override
+	public String getTitle(Locale locale) {
+		return _language.get(locale, "translations");
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemSelectorView.java:60]: Translations??

File: modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemDescriptor.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.fragment.web.internal.item.selector;
+
+import com.liferay.fragment.model.FragmentCollection;
+import com.liferay.item.selector.ItemSelectorViewDescriptor;
```
Comment (ealonso) [inline modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/item/selector/FragmentCollectionItemDescriptor.java:18]: I think you mix two commits

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by ealonso** [comment]
Hey @BarbaraCabrera,
I think we already have an item selector for collections, 



So, I think we need to improve it and reuse, you can find it here fragment-collection-item-selector-web

**Comment by BarbaraCabrera** [comment]
ok-crying

**Comment by ealonso** [comment]
Closing, since we need to send another approach

**Pull request closed without merging**